floodviz/static/js/analytics_helper.js

```diff
@@ -0,0 +1,45 @@
+(function (i, s, o, g, r, a, m) {
+	i['GoogleAnalyticsObject'] = r;
+	i[r] = i[r] || function () {
+			(i[r].q = i[r].q || []).push(arguments)
+		}, i[r].l = 1 * new Date();
+	a = s.createElement(o),
+		m = s.getElementsByTagName(o)[0];
+	a.async = 1;
+	a.src = g;
+	m.parentNode.insertBefore(a, m)
+})(window, document, 'script', 'https://www.google-analytics.com/analytics.js', 'ga');
+
+ga('create', 'UA-103831764-1', 'auto');
+ga('send', 'pageview');
+
+
+// GA for DOM content not related to figures.
+document.addEventListener('DOMContentLoaded', function (event) {
+
+	// Social Media Click events for GA
+	var sm_icons = document.getElementsByClassName('icon');
+	Array.prototype.forEach.call(sm_icons, function (element) {
```

Comment:
sm_icons and links are HTMLCollections which do not have the forEach function available to them.

 I was trying to keep the GA code as separated as possible by putting this logic in this file.